lambda_src/drivers/destination_s3.py

```diff
@@ -82,15 +82,17 @@ def write_to_s3(bucket: Text, filename: Text, content: AnyStr) -> Dict[Text, Any
     )
 
 
-def initialize(destination, batch_id: Text):
+def initialize(destination: Text):
     bucket, prefix = parse_destination_uri(destination)
     content = ''  # We use empty body for creating a folder
-    if not prefix:
-        prefixed_folder_path = f'{DATA_FOLDER_NAME}/{batch_id}/'
-    else:
-        prefixed_folder_path = f'{prefix}/{batch_id}/'
+    prefixed_folder_path = ''
+    if prefix and prefix.count('/') and not prefix.endswith('/'):
```

Comment:
1) `not prefix.endswith('/')` for when the `destination-uri` is: `s3://bucket_name/a/b/c`
2) `prefix.count('/')`  for when the `destination-uri` is: `s3://bucket_name/a`, since we don't want a folder in this case. The `/` from `/a` gets lstripped in line 34, so the count becomes zero.

Reply:
why does this need to happen? why does anything need to happen when initializing an S3 destination?

Reply:
Not 100% certain but from my read of it —

~~~
def initialize(destination, batch_id):
    pass
~~~